Source/WebCore/dom/Document.cpp

```diff
@@ -11599,6 +11601,29 @@ void Document::elementDisconnectedFromDocument(const Element& element)
         m_cachedFirstElementWithAttribute = std::nullopt;
 }
 
+FrameMemoryMonitor& Document::frameMemoryMonitor()
+{
+    ASSERT(!frame()->isMainFrame());
```

Comment:
I remember you said you need to monitor entire page's memory usage. If so, you need to have memoryMonitor for main frame then.

Reply:
we will stick with subframes for now as we don't want to unload the main frame